requester/requester.go

```diff
@@ -77,7 +80,13 @@ func (r *Requester) Run() (Report, error) {
 
 	defer cancel()
 
-	if err := dispatcher.New(numWorker).Do(ctx, maxIter, r.record(req, interval)); err != nil {
+	r.start = time.Now()
+	go r.refreshState()
+
+	switch err := dispatcher.New(numWorker).Do(ctx, maxIter, r.record(req, interval)); err {
+	case nil, context.Canceled, context.DeadlineExceeded:
```

Comment:
Why is `case nil` returning `r.end(err)` and not the same thing as default?

Reply:
Here `default` is the case of an _real_ error that prevents the correct execution of `Do`, so it returns an empty returns and the error.
The three values in this `case` L87 are expected and do not prevent the run (`context.DeadlineExceeded` correspond to `globalTimeout` reached for instance), so ihere we end the process the regular way via `r.end`
Then we only use the value of the error to display the status in the CLI: https://github.com/benchttp/runner/blob/71013d338af1dea57966dbd4029b241e28c66bc6/requester/print.go#L73-L86

Is it clearer? 🤔

Reply:
Ok, thanks for the explanation!
It is clearer!